### PR TITLE
Fix/frontend86 linking pages

### PR DESCRIPTION
**Pull request by Barracuda713**
Add linking between the Main and the Sign-in pages:
- Sign Up -> Sign In
- Sign In (hasToken) -> getUser -> Root
- Sign In -> Root

Add "Confirm password" to Sign Up page

Add form inputs validation for Sign In, Sign Up & Restore password pages

### CODE REVIEW

**Comment by Barracuda713** [comment]
[Add linking between the Main and the Sign-in pages](https://trello.com/c/vaPKbvzd/86-add-linking-between-the-main-and-the-sign-in-pages)

**Comment by Barracuda713** [comment]
[The Sign-up page doesn't have the "Repeat password" input field](https://trello.com/c/RTR4DdsF/84-the-sign-up-page-doesnt-have-the-repeat-password-input-field)

**Comment by Barracuda713** [comment]
[Validation on frontend side for Sign-in / Sign-up + restore password](https://trello.com/c/p9kMwJt9/59-validation-on-frontend-side-for-sign-in-sign-up-restore-password)

File: frontend/src/components/auth/components/sign-up-form/common.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ const DEFAULT_SIGN_UP_PAYLOAD: UserSignUpRequestDto = {
   email: '',
   username: '',
   password: '',
+  password_confirm: '',
```
Comment (droztoff) [inline frontend/src/components/auth/components/sign-up-form/common.ts]: use camelCase

File: frontend/src/components/app/app.tsx [hunk 2]
```diff
@@ -1,26 +1,43 @@
...
+  const hasUser = Boolean(user);
+
   const isHaveHeader = isRouteHaveHeader(pathname);
 
+  useEffect(() => {
+    if (hasToken) {
+      dispatch(authActions.loadCurrentUser());
+    }
+  }, [hasToken, hasUser, dispatch]);
+
```
Comment (foxt451) [inline frontend/src/components/app/app.tsx:33]: if I understand correctly, this dispatch will work twice (user is null initially, then the user changes in the reducer and useEffect works twice)

File: frontend/src/components/auth/components/form-controls.module.scss [hunk 3]
```diff
@@ -7,3 +7,9 @@
   margin-top: 13px;
   margin-bottom: 22px;
 }
+
+.input-error {
+  border: 1px solid var(--input-error-color);
+  border-radius: 3px;
+  box-shadow: 0 0 3px 0 rgb(244 27 59 / 80%), 0 0 5px 0 rgb(244 27 59 / 50%);
```
Comment (foxt451) [inline frontend/src/components/auth/components/form-controls.module.scss]: I guess you could take this color into a variable (and apply sass's color modification function to change opacity, if you like)

**Review by foxt451 — CHANGES_REQUESTED** [review]

**Review by foxt451 — DISMISSED** [review]

**Review by droztoff — DISMISSED** [review]

**Review by droztoff — APPROVED** [review]

**Pull request merged**